source/observables.txt

```diff
@@ -0,0 +1,243 @@
+provided by the ``Observer`` class to print results, handle errors, and
+perform additional data processing.
+
+To learn more about processing results, see the following API documentation:
+
+- `Observable <{+api+}/org/mongodb/scala/Observable.html>`__
+- `Subscription <{+api+}/org/mongodb/scala/Subscription.html>`__
+- `Observer <{+api+}/org/mongodb/scala/Observer.html>`__
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``restaurants`` collection in the ``sample_restaurants``
+database from the :atlas:`Atlas sample datasets </sample-data>`. To access this collection
+from your Scala application, create a ``MongoClient`` that connects to an Atlas cluster
+and assign the following values to your ``database`` and ``collection`` variables:
+
+.. literalinclude:: /includes/observables.scala
+    :language: scala
+    :dedent:
+    :start-after: start-db-coll
+    :end-before: end-db-coll
+
+To learn how to create a free MongoDB Atlas cluster and load the sample datasets, see the
+:atlas:`Get Started with Atlas </getting-started>` guide.
+
+Use Callbacks to Process Results
+--------------------------------
+
+After subscribing to an ``Observable[T]``, you can use the following
+callback methods provided by the ``Observer`` class to access operation results
+or handle errors:
+
+- ``onNext(result: TResult)``: Called when the ``Observer`` receives new results. You
+  can define logic for processing results by overriding this method.
+
+- ``onError(e: Throwable)``: Called when the operation generates an error and prevents
+  the ``Observer`` from receiving more data from the ``Observable``. You can define
+  error handling logic by overriding this method.
+
+- ``onComplete()``: Called when the ``Observer`` has consumed all results from the 
+  ``Observable``. You can perform any final data processing by overriding this method.
+
+The following sections show how to customize these methods to process read and
+write operation results from an ``Observable``.
+
+Access Read Operation Results
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To access data retrieved by a read operation, create an ``Observable[T]``
+to store the operation results. Then, subscribe to the observable and
+override the ``Observer`` class callback methods to process the results.
+
+This example queries the ``restaurants`` database for documents
+in which the ``cuisine`` value is ``"Czech"``. To retrieve
+and process results, the example assigns a ``Observable[Document]``
+to the operation and performs the following actions:
+
+- Calls the ``subscribe()`` method to subscribe to the ``Observable``
+  and passes an ``Observer`` as a parameter
+- Customizes the ``onNext()`` method to print each retrieved document,
+  which are ``Document`` instances
+- Customizes the ``onError()`` method to print any errors
+- Customizes the ``onComplete()`` methods to print a message after all
+  the results from the ``Observable`` are processed
```

Comment:
Would using "Overrides" instead of "Customizes" here be more accurate/descriptive?

```suggestion
- Overrides the ``onNext()`` method to print each retrieved document,
  which are ``Document`` instances
- Overrides the ``onError()`` method to print any errors
- Overrides the ``onComplete()`` methods to print a message after all
  the results from the ``Observable`` are processed
```